src/app/core/services/profile/profile.service.ts

```diff
@@ -26,30 +30,55 @@ export class ProfileService {
     };
     try {
       let data: any = await this.httpService.post(config);
+      let userDetails = await this.localStorage.getLocalData(localKeys.USER_DETAILS);
+      userDetails.user= data.user;
+      await this.localStorage.setLocalData(localKeys.USER_DETAILS, userDetails);
       this.loaderService.stopLoader();
-      this.router.navigate([CommonRoutes.TABS+'/'+CommonRoutes.PROFILE], { queryParams: formData });
+      this.router.navigate([CommonRoutes.TABS + '/' + CommonRoutes.PROFILE], { queryParams: formData });
```

Comment:
do location.back instead of  navigating like this

Reply:
inside location.back, we cant pass query params